Sources/TuistGenerator/Generator/ConfigGenerator.swift

```diff
@@ -251,12 +251,17 @@ final class ConfigGenerator: ConfigGenerating {
         }
 
         // Entitlements
-        if let entitlements = target.entitlements, let path = entitlements.path {
-            let relativePath = path.relative(to: sourceRootPath).pathString
-            if project.xcodeProjPath.parentDirectory == sourceRootPath {
-                settings["CODE_SIGN_ENTITLEMENTS"] = .string(relativePath)
-            } else {
-                settings["CODE_SIGN_ENTITLEMENTS"] = .string("$(SRCROOT)/\(relativePath)")
+        if let entitlements = target.entitlements {
+            if let path = entitlements.path {
+                let relativePath = path.relative(to: sourceRootPath).pathString
+                if project.xcodeProjPath.parentDirectory == sourceRootPath {
+                    settings["CODE_SIGN_ENTITLEMENTS"] = .string(relativePath)
+                } else {
+                    settings["CODE_SIGN_ENTITLEMENTS"] = .string("$(SRCROOT)/\(relativePath)")
+                }
+            }
+            if case let .variable(configName) = entitlements {
+                settings["CODE_SIGN_ENTITLEMENTS"] = .string(configName)
```

Comment:
nit: These lines will override the value that was set above. Is that desired? Should this `if else case let ...`, so it's only run if `entitlements.path` is nil?